### ISSUE TITLE
APPLICATION ERROR: Project file (*.csproj or *.xproj) not found in the project folder and no assembly path found in the config file

### ISSUE THREAD

**Issue by bugproof** [jburzynski/TypeGen#65]
```
PM> typegen -v generate
Generating files for project "."...
No config file found for project ".". Default configuration will be used.
Assembly path not found in the config file. Assembly file will be searched for recursively in the project's bin\.
APPLICATION ERROR: Project file (*.csproj or *.xproj) not found in the project folder and no assembly path found in the config file
   at TypeGen.Cli.Business.ConfigProvider.GetDefaultAssemblyPath(String projectFolder) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Business\ConfigProvider.cs:line 106
   at TypeGen.Cli.Business.ConfigProvider.UpdateConfigAssemblyPaths(TgConfig config, String projectFolder) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Business\ConfigProvider.cs:line 75
   at TypeGen.Cli.Business.ConfigProvider.GetConfig(String configPath, String projectFolder) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Business\ConfigProvider.cs:line 48
   at TypeGen.Cli.Program.Generate(String projectFolder, String configPath) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Program.cs:line 116
   at TypeGen.Cli.Program.Main(String[] args) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Program.cs:line 78
```

The project is there so it looks like a bug

**Comment by jburzynski** [comment]
It probably has something to do with the current working directory.
I think in the PM console you need to specify the project folder like this:

```
PM> typegen -v -p "MyProjectName" generate
```

**Comment by wrathofodin** [comment]
There's progress:

```
Using project assembly found in: XXX\bin\Debug\net461\XXX.exe
GENERIC ERROR: Could not load file or assembly 'XXX, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at TypeGen.Cli.Program.Generate(String projectFolder, String configPath) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Program.cs:line 121
   at TypeGen.Cli.Program.Main(String[] args) in D:\Google Drive\Projects\type-gen\src\TypeGen\TypeGen.Cli\Program.cs:line 78
```

The assembly is a regular .NET Framework 4.6.1 exe

**Comment by jburzynski** [comment]
Can you please tell me what is the type of the project you're running TypeGen on? Is it a web project or something else?

Also can you paste the whole output from the console? Thanks

**Comment by wrathofodin** [comment]
It's a web project - ASP.NET Core specifically.

**Comment by jburzynski** [comment]
So there are 2 things I'm not quite getting - maybe I'm missing something and you could possibly explain some more:

1. The assembly is in the `net461` folder and the project is an ASP.NET Core project - I think normally ASP.NET Core projects would go to `netcoreappX.X` instead of `netXXX` - can you maybe explain more on how you build the project?
2. The assembly is an exe, but I guess normally it would be a dll when building ASP.NET Core projects - are you by chance publishing the project (i.e. not just building it)?

**Comment by wrathofodin** [comment]
@jburzynski ASP.NET Core can target .NET Framework see https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-2.2#aspnet-core-targeting-net-framework

**Comment by jburzynski** [comment]
Cool, that's good to know :)

I tried to replicate this error with an ASP.NET Core 2.1 app (and also ASP.NET Core 2.2 app) targeting .NET Framework 4.6.1, but "it works on my machine".

You could try to point to the .NET Core assembly in tgconfig.json and see if this would work:

```json
{
  "assemblies": ["bin/Debug/netcoreapp2.x/XXX.dll"]
}
```

If you had the possibility and would like to share this information, you could also specify which external dependencies are you using for your project, which would help in replicating the problem.

I also noticed that TypeGen.Cli is a .NET Core 2.1 app and could be upgraded to .NET Core 2.2 (which I will do for the next release), but this, in theory, should not be the problem here, as the assembly is .NET Framework 4.6.1. Just in case, if your web application is ASP.NET Core 2.2, you could try it with the new release which I will do in a couple of days.

**Comment by wrathofodin** [comment]
Unfortunately no luck.

**Comment by jburzynski** [comment]
Ok, I'll look into that in more detail.

Yes, please open a separate issue - I'll aim to add struct support for the next release.

**Comment by jburzynski** [comment]
But actually, for the error with System.TimeSpan, there is a different solution. You'll probably need to annotate the TimeSpan property with `TsTypeAttribute` or add `.TsType()` in the generation spec, e.g.:

```csharp
class MyClass
{
  [TsType(TsType.Number)]
  public TimeSpan MyProperty { get; set; }
}
```

But I'll add struct support anyway, maybe someone would need to generate a struct.

**Comment by jburzynski** [comment]
Also, if you'd like to map TimeSpan to some TypeScript type by default, you can use the `customTypeMappings` CLI parameter.

**Comment by jburzynski** [comment]
One more question regarding this message:
> Unfortunately no luck. It says "no assembly path found in the config file" even if it's there.

Can you make sure that you don't have anything specified in the `assemblyPath` parameter? When I try it with `assemblies` specified, it reads assemblies from the `assemblies` parameter.

Can you also check if it loads the correct config file (it should output the currently used config with the `verbose` option)?

**Comment by jburzynski** [comment]
I just released version 2.4.0, which updates TypeGen CLI to .NET Core 2.2 - you can try if it works with the new CLI.

I'm not able to replicate this error, so I'm going to investigate some more if more people report the same issue. At the moment I guess the solution would be to use the programmatical API.

**Comment by viniciusverasdossantos** [comment]
I think I found the problem. Occurs when the assembly name is different from the project name.
